src/CodeReview.Orchestrator/Utils/OptionMetadataProvider.cs

```diff
@@ -7,15 +7,34 @@
 
 namespace GodelTech.CodeReview.Orchestrator.Utils
 {
-    public class CommandLineJsonConvert : ICommandLineJsonConvert
+    public class OptionMetadataProvider : IOptionMetadataProvider
     {
-        public string Serialize(params Type[] type)
+        private readonly ICommandLineOptionsTypeProvider _commandLineOptionsTypeProvider;
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
+
+        public OptionMetadataProvider(ICommandLineOptionsTypeProvider commandLineOptionsTypeProvider)
         {
-            if (type is null) throw new ArgumentNullException(nameof(type));
+            _commandLineOptionsTypeProvider = commandLineOptionsTypeProvider ?? throw new ArgumentNullException(nameof(commandLineOptionsTypeProvider));
+
+            _jsonSerializerSettings = new JsonSerializerSettings
+            {
+                NullValueHandling = NullValueHandling.Ignore, 
+                Formatting = Formatting.Indented
+            };
+        }
+
+        public string GetOptionsMetadata(Assembly assembly = null)
```

Comment:
`assembly = null` doesn't make a lot of sense because nobody passes assembly as parameter. I think this method may be simplified by parameter removal.